### PR TITLE
{bio}[foss/2015b] Sambamba 0.5.9 (WIP)

### PR DESCRIPTION
**Pull request by nathanhaigh**
Initial commit of Sambamba easyconfig.

The project has several dependencies which are integrated into the source code via git submodules. As a result, the downloaded source for Sambamba doesn't include these. In addition, building of Sambamba also attempts to build the dependencies in the subdirectories.

I'm in the process of making easyconfigs for the dependencies. Dependencies requiring easyconfigs:

| Dependency | EasyConfig Status | Build Info |
| --- | --- | --- |
| LZ4 v131 | Done |  |
| HTSlib v1.3 | Done |  |
| D Lang | Not started | https://wiki.dlang.org/GDC/Installation/Generic |
| BioD | Not started |  |

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+name = 'LZ4'
+version = '131'
+
+homepage = 'https://github.com/Cyan4973/lz4'
+description = """Extremely fast lossless compression algorithm."""
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+toolchainopts = {}  # toolchain options, e.g. opt, pic, usempi, optarch, ...
+
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
```
Comment (boegel) [inline easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb]: please drop this comment?

File: easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'MakeCp'
+
+name = 'LZ4'
+version = '131'
+
+homepage = 'https://github.com/Cyan4973/lz4'
+description = """Extremely fast lossless compression algorithm."""
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+toolchainopts = {}  # toolchain options, e.g. opt, pic, usempi, optarch, ...
```
Comment (boegel) [inline easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb]: this is pretty useless?

File: easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+patches = []
+
+dependencies = []
+
+# make sure compilation flags are passed down (e.g. to enable OpenMP)
+buildopts = ' CC="$CXX" CCFLAGS="$CPPFLAGS $CXXFLAGS"'
+
+files_to_copy = [
+  (['programs/lz4'], 'bin'),
+  (['lib', 'examples'], ''),
```
Comment (boegel) [inline easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb]: please fix indent

File: easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+]
+
+# The sanity test MUST be tuned before going production and submitting your contribution to upstream git repositories
```
Comment (boegel) [inline easybuild/easyconfigs/l/LZ4/LZ4-131-foss-2015b.eb]: please drop comment?

File: easybuild/easyconfigs/s/Sambamba/Sambamba-0.5.9-foss-2015b.eb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+patches = []
+
+dependencies = [
+    ('HTSlib', '1.3'),
+    ('LZ4', '131'),
+# D programming language: http://dlang.org/
+# BioD: https://github.com/biod/BioD
+]
+
+# The sanity test MUST be tuned before going production and submitting your contribution to upstream git repositories
```
Comment (boegel) [inline easybuild/easyconfigs/s/Sambamba/Sambamba-0.5.9-foss-2015b.eb]: please drop comment

File: easybuild/easyconfigs/s/Sambamba/Sambamba-0.5.9-foss-2015b.eb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+version = '0.6.0'
+
+homepage = 'http://lomereiter.github.io/sambamba'
+description = """Sambamba is a high performance modern robust and fast tool (and library), written
+  in the D programming language, for working with SAM and BAM files."""
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+toolchainopts = {}  # toolchain options, e.g. opt, pic, usempi, optarch, ...
+
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
```
Comment (boegel) [inline easybuild/easyconfigs/s/Sambamba/Sambamba-0.5.9-foss-2015b.eb]: please drop comment

File: easybuild/easyconfigs/s/Sambamba/Sambamba-0.5.9-foss-2015b.eb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+name = 'Sambamba'
+version = '0.6.0'
+
+homepage = 'http://lomereiter.github.io/sambamba'
+description = """Sambamba is a high performance modern robust and fast tool (and library), written
+  in the D programming language, for working with SAM and BAM files."""
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+toolchainopts = {}  # toolchain options, e.g. opt, pic, usempi, optarch, ...
```
Comment (boegel) [inline easybuild/easyconfigs/s/Sambamba/Sambamba-0.5.9-foss-2015b.eb]: useless as is?

**Comment by boegel** [comment]
@nathanhaigh: for the remaining two deps, consider tackling those in a separate PR if they involve more than one easyconfig file; if not, including them here is fine (5 easyconfigs in total in a single PR is manageable)

**Comment by nathanhaigh** [comment]
@boegel - I don't have much experience building compilers, so in terms of building the D langauage compiler, it looks like I'll need to recompile GCC (https://wiki.dlang.org/GDC/Installation/Generic#Building) with `--enable-languages=d` among other things. How should this fit in with EB toolchains and the like?

**Comment by boegel** [comment]
Oh boy, this looks like an adventure for sure. :)

The GCC easyblock we have supports specifying the set of languages to compile for:

```
$ eb -a -e EB_GCC | grep lang
languages*              List of languages to build GCC for (--enable-languages) [default: []]
```

Usually we include this in the GCC easyconfigs:

```
languages = ['c', 'c++', 'fortran']
```

So, it looks like you should look into a `GDC` easyconfig that includes this:

```
easyblock = 'EB_GCC'
name = 'GDC'
...
languages = ['d']
```

I'm not sure if you'll actually get to a working GDC like this, but it's definitely worth a try.
We may have to add a way to disable the 3-stage bootstrap the GCC easyblock now always perform though, or maybe we should look into a `GDC` specific easyblock that builds on top of the existing `GCC` easyblock.

Thoughts @wpoely86?

**Comment by wpoely86** [comment]
It's uncharted territory. Might work, might not.

**Comment by migueldiascosta** [comment]
@nathanhaigh, this PR is being closed for the following reason(s): no activity for > 6 months, obsoleted by more recent PRs.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**